tide/master/rest/FileReceiver.cpp

```diff
@@ -165,7 +161,7 @@ FileReceiver::_handleUpload( const QString& filePath, const std::string& path,
     emit open( filePath, openPromise );
     const bool success = openPromise->get_future().get();
 
-    _server.remove( "tide/upload/" + path );
+    _preparedPaths.removeOne( filePath );
```

Comment:
can be moved right after file.close(), no need to wait for open() to succeed if it is removed always